model/SimulationRunner.py

```diff
@@ -0,0 +1,172 @@
+# This Python file uses the following encoding: utf-8
+from PyQt5.QtCore import pyqtSignal, pyqtSlot, QThread, QMutex, QMutexLocker
+import subprocess
+import os
+import enum
+import logging
+import threading
+import queue
+
+def enqueueOutStream(output, q):
+    while True:
+        line = output.readline()
+        if not line:
+            break
+        q.put(line)
+
+class SimulationRunner(QThread):
+    class InitState(enum.Enum):
+        NONE = "None"
+        INIT = "Initializing"
+        STATED = "Stated"
+        PARSED_ARGS = "Parsed args"
+        LOADING_PARAMS = "loading population and setting up parameters"
+        STARTING_SIM = "starting simulation"
+
+    class SimulationState(enum.Enum):
+        SIMULATION_ONGOING = "Simulation ongoing"
+        STOPPED = "Stopped"
+        DONE = "Done"
+
+    pathToCLI = ""
+    openedFilePath = ""
+    outputDaily = ""
+    outputSummary = ""
+    outputParamsDump = ""
+    outputRunDumpPrefix = ""
+    __currentState = InitState.NONE
+    __currentProgress = 0
+    __process = None
+    __mutex = QMutex()
+    __isThreadStopped = False
+
+    printSimulationMsg = pyqtSignal(str, arguments=["msg"])
+    notifyState = pyqtSignal(str, arguments=["description"])
+    notifyProgress = pyqtSignal(float, arguments=["progress"])
+    clearLog = pyqtSignal()
+
+    def __init__(self):
+        QThread.__init__(self)
+        self.notifyProgress.connect(self.__saveCurrentProgress)
+
+    def __del__(self):
+        self.wait()
+
+    @pyqtSlot(result=bool)
+    def isRunning(self):
+        return (isinstance(self.__currentState, SimulationRunner.InitState) and \
+            self.__currentState != SimulationRunner.InitState.NONE) or \
+            (isinstance(self.__currentState, SimulationRunner.SimulationState) and \
+            self.__currentState == SimulationRunner.SimulationState.SIMULATION_ONGOING)
+
+    def __createCommand(self):
+        cliname = os.path.basename(self.pathToCLI)
```

Comment:
why the basename?

Reply:
I don't really understand the question. What do you think is wrong with it?